### ISSUE TITLE
[Deeplab v3+], The relation between num_shards and training batch size

### ISSUE THREAD

**Issue by sunformoon** [tensorflow/models#4630]
Hi Jay, George, and Yukun,

It there any rule I should follow when I set the num_shards and training batch size? Like, the num_clones is set for how many GPUs I am using.

 In the given example of local_test.sh and build_voc2012_data.py, the batch size is set to be 4 and the _NUM_SHARDS is set to be 4 as well.

Best,
Zhuotun

------------------------

### System information
- **What is the top-level directory of the model you are using**: deeplab
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu16.04
- **TensorFlow installed from (source or binary)**: 
- **TensorFlow version (use command below)**: tensorflow-gpu 1.7.0 
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**: CUDA 9.0/ cuDNN 7
- **GPU model and memory**:16GB V100
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by YknZhu** [comment]
we usually use the largest possible batch size which fits GPU memory :p 

For number of shard, it doesn't matter too much. Larger values potentially gives you slightly better shuffling and then you ends up with many smaller files.

**Comment by sunformoon** [comment]
@YknZhu Yukun, thanks for your prompt response. ;)
I am gonna use a large batch size to fit the GPU then.
